lists/python/test_linked_list.py

```diff
@@ -242,107 +246,114 @@ def test_cl_construct_empty_list():
 def test_cl_add_to_empty_list():
     # arrange
     subject = CircularList()
+    nodeA = Node('a')
 
     #act
-    actual = subject.add(Node('a'))
+    actual = subject.add(nodeA)
 
     # assert
-    assert actual.value == 'a'
+    assert actual == node
     assert subject == ['a']
 
 
 def test_cl_add_to_populated_list():
     # arrange
     subject = CircularList()
     subject.add(Node('a'))
+    nodeB = Node('b')
 
     #act
-    actual = subject.add(Node('b'))
+    actual = subject.add(nodeB)
 
     # assert
-    assert actual.value == 'b'
+    assert actual == nodeB
     assert subject == ['a', 'b']
 
-
+#21
 def test_cl_add_to_list_of_2_plus():
     # arrange
     subject = CircularList()
     subject.add(Node('a'))
     subject.add(Node('b'))
+    nodeC = Node('c')
     
     #act
-    actual = subject.add(Node('c'))
+    actual = subject.add(nodeC)
 
     # assert
-    assert actual.value == 'c'
+    assert actual == nodeC
     assert subject.as_list() == ['a', 'b', 'c']
     assert subject == ['a', 'b', 'c']
 
 
 def test_cl_remove_empty_list():
     # arrange
     subject = CircularList()
+    node = Node('dne')
 
     # assert
-    assert subject.remove('dne') is None
+    assert subject.remove(node) is None
 
 
 def test_cl_remove_only_item():
     # arrange
     subject = CircularList()
-    subject.add(Node('a'))
+    nodeA = Node('a')
+    subject.add(nodeA)
 
     #act
-    actual = subject.remove('a')
+    actual = subject.remove(NodeA)
 
     # assert
-    assert actual.value == 'a'
+    assert actual == NodeA
     assert subject == []
 
 
 def test_cl_remove_first_item_in_list():
     # arrange
     subject = CircularList()
-    subject.add(Node('a'))
+    nodeA = Node('a')
+    subject.add(NodeA)
     subject.add(Node('b'))
 
     #act
-    actual = subject.remove('a')
+    actual = subject.remove(nodeA)
 
     # assert
-    assert actual.value == 'a'
+    assert actual== nodeA
     assert subject == ['b']
 
 
 def test_cl_remove_last_item_in_list():
     # arrange
     subject = CircularList()
     subject.add(Node('a'))
-    subject.add(Node('b'))
+    nodeB = Node('b')
+    subject.add(nodeB)
 
     #act
-    actual = subject.remove('b')
+    actual = subject.remove(NodeB)
 
     # assert
-    assert actual.value == 'b'
+    assert actual == nodeB
     assert subject == ['a']
 
-
 def test_cl_remove_item_from_middle_of_list():
     # arrange
     subject = CircularList()
     subject.add(Node('a'))
     subject.add(Node('b'))
     subject.add(Node('c'))
-    subject.add(Node('d'))
+    nodeD = Node('d')
+    subject.add(nodeD)
     subject.add(Node('e'))
     subject.add(Node('f'))
 
     #act
-    actual = subject.remove('d')
+    actual = subject.remove(nodeD)
 
     # assert
-    assert actual.value == 'd'
+    assert actual == nodeD
```

Comment:
I added a comment on your other PR but the real value in asserting on the node is being able to asserts the `next` and `prev` nodes are what is expected.

Reply:
Is this what you had in mind?

def test_sl_add_to_empty_list():
    # arrange
    subject = LinkedList()
    node = Node('a')

    # act
    actual = subject.add(node)
    next_node = actual.next
    previous_node = actual.prev

    # assert
    assert actual == node
    assert next_node == None
    assert previous_node == None
    assert subject == ['a']